### ISSUE TITLE
Feature request: when more than one attribute with the same name applies to a single element, throw an exception instead of just using the last one

### ISSUE THREAD

**Issue by Dretch** [connect-group/thymesheet#13]
Consider Thymesheet files with duplicate attributes, like:

```
a {
  th-class: "firstClass";
  th-class: "secondClass";
}
```

or, even something like:
```
a {
  th-class: "firstClass";
}
* {
  th-class: "secondClass";
}
```

More than once I have seen people expect this sort of thing to work (by expecting ThymeLeaf/Sheet to combine the attributes), but instead only the last attribute that applies to an element will be used, and the others ignored.

I think both of the above cases should result in exceptions (and they will probably require different bits of code to be changed to implement each of them). Do you agree?

**Comment by 4dz** [comment]
The idea is it works like css. 
If you do this it’s perfectly valid, and it could happen as a result of two or more selectors  cascading (combining) too — which is your second example. 

This is no different than having a css 

* {
color:blue
}

a {
color:red
}

 (For example)

These are perfectly valid scenarios - especially the second one you gave - and changing it would be a breaking change, causing unexpected errors.

**Comment by Dretch** [comment]
I agree that the behaviour makes sense in that it is consistent with CSS, and that changing it will cause unexpected errors.

I checked what errors would occur in our project by hacking Thymesheet to throw some exceptions in the cases I outlined above... and I found quite a few errors when running our project's tests.

Unfortunately perhaps half of these exceptions are thrown because of Thymesheet that is obviously erroneus (especially with the first case, but also sometimes in the second case). It seems pretty common for people to use multiple `th:attr`, `th:if` or `th:classappend` attributes, and expect them to be applied cumulatively.

So, while this behaviour does make sense. I don't think it is ideal because it also results in problems going unnoticed.
